packages/neuron-wallet/src/channel/index.ts

```diff
@@ -70,18 +71,91 @@ export class Listeners {
     })
   }
 
+  static checkPassword = (walletID: string, password: string) => {
+    const myWallet = wallets().find(wallet => wallet.id === walletID)
+    if (!myWallet) {
+      return {
+        status: ResponseStatus.Success,
+        result: false,
+        msg: 'Wallet not find',
+      }
+    }
+    if (myWallet.password === password) {
+      return {
+        status: ResponseStatus.Success,
+        result: true,
+      }
+    }
+    return {
+      status: ResponseStatus.Success,
+      result: false,
+      msg: 'Wrong password',
+    }
+  }
+
+  /**
+   * @static getWallets
+   * @memberof ChannelListeners
+   * @description channel to get wallets
+   */
+  static checkWalletPassword = () => {
+    return ipcMain.on(
+      Channel.CheckWalletPassword,
+      (e: Electron.Event, { walletID, password }: { walletID: string; password: string }) => {
+        e.sender.send(Channel.CheckWalletPassword, Listeners.checkPassword(walletID, password))
+      },
+    )
+  }
+
   /**
    * @static deleteWallet
    * @memberof ChannelListeners
    * @description channel to delete wallet
    */
   static deleteWallet = () => {
-    return ipcMain.on(Channel.DeleteWallet, (e: Electron.Event, address: string) => {
-      e.sender.send(Channel.DeleteWallet, {
-        status: ResponseStatus.Success,
-        reult: `wallet of ${address} deleted`,
-      })
-    })
+    return ipcMain.on(
+      Channel.DeleteWallet,
+      (e: Electron.Event, { walletID, password }: { walletID: string; password: string }) => {
+        const result = Listeners.checkPassword(walletID, password)
```

Comment:
The listener should not be used as a utility function.

Reply:
get it